src/components/PokemonDetails/PokemonDetails.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+import "./PokemonDetails.css";
+
+const PokemonDetails = ({ foundPokemon, getPokemonImage }) => {
+  const [pokemonDetails, setPokemonDetails] = useState([]);
+  const [error, setError] = useState("");
+
+  let pokemonId = foundPokemon[0].url.replace(/\D/g, "").slice(1)
+  const pokemonImage = getPokemonImage(pokemonId);
+
+//   let pokemonImage =  pokemonDetails.sprites.other.dream_world.front_default
+  const getPokemonDetails = async () => {
+
+    // let pokemonId = foundPokemon[0].url.replace(/\D/g, "").slice(1);
+    let pokemonName = foundPokemon[0].name
+    const url = `https://pokeapi.co/api/v2/pokemon/${pokemonName}`;
+
+    try {
+      const res = await fetch(url);
+      const pokemonDetails = await res.json();
+      setPokemonDetails(pokemonDetails);
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+// for some reason I don't know yet --it is not reseting the state when we type a diferent name or id 
+// i'm not sure if only clearing the search it is what we need, probably we need a way to look into the details again..
+  useEffect(() => {
+    getPokemonDetails();
+  }, []);
+
```

Comment:
Even with the error, I'm glad we found ways to merge the code, and I am happy about where it's going.